docs/user-guides/acme-dns-validation.md

```diff
@@ -0,0 +1,94 @@
+# Issuing an ACME certificate using DNS validation
+
+`cert-manager` can be used to obtain certificates from a CA using the [ACME][1] protocol. The ACME protocol supports various challenges which are used to prove ownership of a domain so that a valid certificate can be issued for that domain. One such challenge is `dns-01`. With the DNS challenge, you prove you own a domain by proving you control domain's DNS records. You do this by creating a TXT record with specific content.
+
+The following `Issuer` defines the necessary information to enable DNS validation. You can read more about the `Issuer` resource [here][2].
+
+```yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-staging
+  namespace: default
+spec:
+  acme:
+    # The ACME server URL
+    server: https://acme-staging.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: user@example.com
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      name: letsencrypt-staging
+    # ACME dns-01 provider configurations
+    dns01:
+      # Here we define a list of DNS-01 providers that can solve DNS challenges
+      providers:
+      - name: prod-dns
+        clouddns:
+          # A secretKeyRef to a google cloud json service account
+          serviceAccountSecretRef:
+            name: clouddns-service-account
+            key: service-account.json
+          # The project in which to update the DNS zone
+          project: gcloud-prod-project
+      - name: cf-dns
+        cloudflare:
+          email: user@example.com
+          # A secretKeyRef to a cloudflare api key
+          apiKeySecretRef:
+            name: cloudflare-api-key
+            key: api-key.txt
+```
```

Comment:
Worth mentioning that we have defined two DNS providers here, and explaining why and what that means (i.e. explaining that we can define multiple and why we would do this, and then linking out to the spec docs that explain this a bit more maybe?)